source/common/grpc/context_impl.cc

```diff
@@ -94,5 +99,18 @@ ContextImpl::resolveDynamicServiceAndMethod(const Http::HeaderEntry* path) {
   return RequestStatNames{service, method};
 }
 
+std::pair<absl::optional<ContextImpl::RequestStatNames>, std::unique_ptr<std::string>>
+ContextImpl::resolveDynamicServiceAndMethodWithDotReplaced(const Http::HeaderEntry* path) {
+  absl::optional<Common::RequestNames> request_names = Common::resolveServiceAndMethod(path);
+  if (!request_names) {
+    return {};
+  }
+  auto service_name =
+      std::make_unique<std::string>(absl::StrReplaceAll(request_names->service_, {{".", "_"}}));
```

Comment:
Had to create a new string and pass return it in the pair because StrReplaceAll() returns new strings. The RequestStatNames only store string_view which is a weak reference and only a view of the original string.